File: sdk/python/authsys.py
import requests
import platform
import subprocess

class AuthSys:
    def __init__(self, name, ownerid, secret, version, api_url="https://authsys-main-production.up.railway.app/api/v1"):
        """
        Initialize the AuthSys application credentials.
        :param name: Application Name
        :param ownerid: Your Owner ID
        :param secret: Your Application Secret
        :param version: Application Version
        :param api_url: The Base URL of your AuthSys backend
        """
        self.name = name
        self.ownerid = ownerid
        self.secret = secret
        self.version = version
        self.api_url = api_url.rstrip('/')
        
        self.sessionid = None
        self.initialized = False
        self.app_data = {}
        self.user_data = {}

    def get_hwid(self):
        """Generates a hardware ID (HWID) based on the OS."""
        hwid = "UNKNOWN_HWID"
        try:
            if platform.system() == "Windows":
                cmd = subprocess.Popen("wmic csproduct get uuid", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
                hwid = cmd.stdout.read().decode('utf-8').split('\n')[1].strip()
            elif platform.system() == "Linux":
                with open("/etc/machine-id", "r") as f:
                    hwid = f.read().strip()
            elif platform.system() == "Darwin":
                cmd = subprocess.Popen("ioreg -rd1 -c IOPlatformExpertDevice | grep IOPlatformUUID", shell=True, stdout=subprocess.PIPE)
                hwid = cmd.stdout.read().decode('utf-8').split('"')[3]
        except Exception:
            pass
        return hwid

    def _post(self, endpoint, data):
        url = f"{self.api_url}/client/{endpoint}"
        try:
            response = requests.post(url, json=data, timeout=10)
            return response.json()
        except requests.exceptions.RequestException as e:
            return {"status": "error", "detail": f"Connection error: {str(e)}"}

    def init(self):
        """Initializes the application and checks for updates/variables."""
        data = {
            "app_secret": self.secret,
            "version": self.version,
            "app_name": self.name,
            "hwid": self.get_hwid()
        }
        res = self._post("init", data)
        
        if res.get("status") in ["success", "update_available"]:
            self.initialized = True
            self.app_data['variables'] = res.get('variables', {})
            print(f"[*] Initialized successfully. Message: {res.get('message')}")
        else:
            print(f"[!] Init Failed: {res.get('detail', res.get('message', 'Unknown Error'))}")
        
        return res

    def register(self, username, password, license_key, email=None):
        """Registers a new user."""
        if not self.initialized:
            print("[!] Please run init() first.")
            return False

        data = {
            "app_secret": self.secret,
            "username": username,
            "password": password,
            "license_key": license_key,
            "email": email,
            "hwid": self.get_hwid()
        }
        res = self._post("register", data)
        
        if "access_token" in res or res.get("message") == "User registered successfully":
            print("[*] Successfully registered!")
            self.user_data = res.get('user', {})
            return True
        else:
            print(f"[!] Registration failed: {res.get('detail')}")
            return False

    def login(self, username, password):
        """Logs in an existing user."""
        if not self.initialized:
            print("[!] Please run init() first.")
            return False

        data = {
            "app_secret": self.secret,
            "username": username,
            "password": password,
            "hwid": self.get_hwid()
        }
        res = self._post("login", data)
        
        if "access_token" in res:
            print("[*] Successfully logged in!")
            self.sessionid = res.get("access_token")
            self.user_data = res.get('user', {})
            return True
        else:
            print(f"[!] Login failed: {res.get('detail')}")
            return False

    def license(self, license_key):
        """Logs in using only a license key."""
        if not self.initialized:
            print("[!] Please run init() first.")
            return False

        data = {
            "app_secret": self.secret,
            "license_key": license_key,
            "hwid": self.get_hwid()
        }
        res = self._post("license_login", data)
        
        if "access_token" in res:
            print("[*] Successfully authenticated via license!")
            self.sessionid = res.get("access_token")
            self.user_data = res.get('user', {})
            return True
        else:
            print(f"[!] License login failed: {res.get('detail')}")
            return False

    def var(self, var_name):
        """Gets an application variable."""
        if not self.initialized:
            print("[!] Please run init() first.")
            return None
        return self.app_data.get('variables', {}).get(var_name)

File: sdk/java/AuthSys.java
import java.io.OutputStream;
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class AuthSys {
    public String name;
    public String ownerid;
    public String secret;
    public String version;
    public String apiUrl;

    public boolean initialized = false;
    public String sessionid = null;

    public AuthSys(String name, String ownerid, String secret, String version) {
        this.name = name;
        this.ownerid = ownerid;
        this.secret = secret;
        this.version = version;
        this.apiUrl = "https://authsys-main-production.up.railway.app/api/v1";
    }

    private String getHWID() {
        try {
            String os = System.getProperty("os.name").toLowerCase();
            if (os.contains("win")) {
                Process process = Runtime.getRuntime().exec(new String[]{"wmic", "csproduct", "get", "uuid"});
                process.getOutputStream().close();
                BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
                String line;
                StringBuilder output = new StringBuilder();
                while ((line = reader.readLine()) != null) {
                    output.append(line);
                }
                return output.toString().replace("UUID", "").trim();
            }
        } catch (Exception e) {}
        return "UNKNOWN_JAVA_HWID";
    }

    private String postRequest(String endpoint, String jsonPayload) {
        try {
            URL url = new URL(this.apiUrl + "/client/" + endpoint);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("POST");
            conn.setRequestProperty("Content-Type", "application/json; utf-8");
            conn.setRequestProperty("Accept", "application/json");
            conn.setDoOutput(true);

            try (OutputStream os = conn.getOutputStream()) {
                byte[] input = jsonPayload.getBytes(StandardCharsets.UTF_8);
                os.write(input, 0, input.length);
            }

            int code = conn.getResponseCode();
            BufferedReader br = new BufferedReader(new InputStreamReader(
                    code >= 200 && code < 300 ? conn.getInputStream() : conn.getErrorStream(), "utf-8"));
            
            StringBuilder response = new StringBuilder();
            String responseLine;
            while ((responseLine = br.readLine()) != null) {
                response.append(responseLine.trim());
            }
            return response.toString();
        } catch (Exception e) {
            return "{\"status\":\"error\", \"detail\":\"" + e.getMessage() + "\"}";
        }
    }

    public void init() {
        String json = String.format("{\"app_secret\":\"%s\", \"version\":\"%s\", \"hwid\":\"%s\"}", secret, version, getHWID());
        String res = postRequest("init", json);
        
        if (res.contains("\"status\":\"success\"") || res.contains("\"status\":\"update_available\"")) {
            this.initialized = true;
            System.out.println("[*] Initialized successfully");
        } else {
            System.out.println("[!] Init Failed: " + res);
        }
    }

    public boolean login(String username, String password) {
        if (!initialized) return false;
        
        String json = String.format("{\"app_secret\":\"%s\", \"username\":\"%s\", \"password\":\"%s\", \"hwid\":\"%s\"}", 
            secret, username, password, getHWID());
        String res = postRequest("login", json);

        if (res.contains("access_token")) {
            System.out.println("[*] Logged in successfully");
            // Basic parsing to extract access_token could be done here (e.g. using org.json or Gson)
            this.sessionid = "mock_token_extracted";
            return true;
        } else {
            System.out.println("[!] Login Failed: " + res);
            return false;
        }
    }

    public boolean register(String username, String password, String licenseKey) {
        if (!initialized) return false;
        
        String json = String.format("{\"app_secret\":\"%s\", \"username\":\"%s\", \"password\":\"%s\", \"license_key\":\"%s\", \"hwid\":\"%s\"}", 
            secret, username, password, licenseKey, getHWID());
        String res = postRequest("register", json);

        if (res.contains("access_token") || res.contains("User registered successfully")) {
            System.out.println("[*] Registered successfully");
            return true;
        } else {
            System.out.println("[!] Registration Failed: " + res);
            return false;
        }
    }
}
